services/management/management.go

```diff
@@ -38,9 +56,7 @@ type Service struct {
 	topologyConsumer TopologyConsumer
 
 	sync.RWMutex
```

Comment:
the mutex belong out here only if `data` is immutable (which it is - only replaced in it's entirety from the adapter). if we wanted to make changes to the struct internally the mutex would have to be inside the struct..

it would be nice if we had a simple way to indicate `data` is immutable. one way to deal with the fact that we don't is to name it using the word `cache` so maybe something like `cachedManagementData`?

Reply:
the data is always replaced as whole ... so that is why mutex is outside. i don't like to add word cached (because we don't allow direct access anyway) and i don't like to have management twice.